lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -714,6 +748,21 @@ def main():
         else:
             module.fail_json(msg="Bucket parameter is required.")
 
+    # Copy an object from a bucket, not the entire bucket
+    if mode == 'copyobj':
+        if obj is None:
+            module.fail_json(msg="object parameter is required")
+        if src_obj is None:
+            module.fail_json(msg="src_object parameter is required")
+        if src_bucket is None:
+            src_bucket = bucket
+        if bucket:
+            copyrtn = copy_key(module, s3, bucket, src_bucket, obj, src_obj)
+            if copyrtn is True:
```

Comment:
It looks like copy_key calls exit_json so you don't need this line and the one following.